// oci-keymanagement/index.js
"use strict";
/**
 * Vault Key Management API
 * Use the Key Management API to manage vaults and keys. For more information, see [Managing Vaults](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingvaults.htm) and [Managing Keys](https://docs.oracle.com/iaas/Content/KeyManagement/Tasks/managingkeys.htm).

 * OpenAPI spec version: release
 * Contact: sparta_kms_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.KmsVaultWaiter = exports.KmsVaultClient = exports.KmsManagementWaiter = exports.KmsManagementClient = exports.KmsHsmClusterWaiter = exports.KmsHsmClusterClient = exports.KmsCryptoClient = exports.EkmWaiter = exports.EkmClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const ekm_waiter = __importStar(require("./lib/ekm-waiter"));
const kmshsmcluster_waiter = __importStar(require("./lib/kmshsmcluster-waiter"));
const kmsmanagement_waiter = __importStar(require("./lib/kmsmanagement-waiter"));
const kmsvault_waiter = __importStar(require("./lib/kmsvault-waiter"));
exports.EkmClient = client.EkmClient;
exports.EkmWaiter = ekm_waiter.EkmWaiter;
exports.KmsCryptoClient = client.KmsCryptoClient;
exports.KmsHsmClusterClient = client.KmsHsmClusterClient;
exports.KmsHsmClusterWaiter = kmshsmcluster_waiter.KmsHsmClusterWaiter;
exports.KmsManagementClient = client.KmsManagementClient;
exports.KmsManagementWaiter = kmsmanagement_waiter.KmsManagementWaiter;
exports.KmsVaultClient = client.KmsVaultClient;
exports.KmsVaultWaiter = kmsvault_waiter.KmsVaultWaiter;
