// oci-sdk/index.js
/**
 * Copyright (c) 2020, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfigFileAuthenticationDetailsProvider = exports.LOG = exports.ConvertStringToType = exports.HandleErrorResponse = exports.OciError = exports.DefaultRequestSigner = exports.signer = exports.SimpleAuthenticationDetailsProvider = void 0;
// Common
exports.common = require("oci-common");
exports.SimpleAuthenticationDetailsProvider = exports.common.SimpleAuthenticationDetailsProvider;
exports.signer = exports.common.DefaultRequestSigner;
exports.DefaultRequestSigner = exports.common.DefaultRequestSigner;
exports.OciError = exports.common.OciError;
exports.HandleErrorResponse = exports.common.handleErrorResponse;
exports.ConvertStringToType = exports.common.convertStringToType;
exports.LOG = exports.common.LOG;
exports.ConfigFileAuthenticationDetailsProvider = exports.common.ConfigFileAuthenticationDetailsProvider;
// Analytics Service
exports.analytics = require("oci-analytics");
// Announcement Service
exports.announcementsservice = require("oci-announcementsservice");
// API Gateway Service
exports.apigateway = require("oci-apigateway");
// Audit Service
exports.audit = require("oci-audit");
// Autoscaling Service
exports.autoscaling = require("oci-autoscaling");
// Budget Service
exports.budget = require("oci-budget");
// Container Engine
exports.containerengine = require("oci-containerengine");
// Core
exports.core = require("oci-core");
// Database
exports.database = require("oci-database");
// Dataflow Service
exports.dataflow = require("oci-dataflow");
// Data Science Service
exports.datascience = require("oci-datascience");
// Data Catalog Service
exports.datacatalog = require("oci-datacatalog");
// DNS
exports.dns = require("oci-dns");
// DTS
exports.dts = require("oci-dts");
// Email
exports.email = require("oci-email");
// Events
exports.events = require("oci-events");
// File Storage
exports.filestorage = require("oci-filestorage");
//Functions
exports.functions = require("oci-functions");
//Health checks
exports.healthchecks = require("oci-healthchecks");
// Identity
exports.identity = require("oci-identity");
// Integration  Service
exports.integration = require("oci-integration");
// Key-Management
exports.keymanagement = require("oci-keymanagement");
// Limits
exports.limits = require("oci-limits");
// Load Balancer
exports.loadbalancer = require("oci-loadbalancer");
// Marketplace
exports.marketplace = require("oci-marketplace");
// Monitoring
exports.monitoring = require("oci-monitoring");
// OCE
exports.oce = require("oci-oce");
// Object Storage
exports.objectstorage = require("oci-objectstorage");
// ODA
exports.oda = require("oci-oda");
// ONS
exports.ons = require("oci-ons");
// OS Management
exports.osmanagement = require("oci-osmanagement");
// Resource manager
exports.resourcemanager = require("oci-resourcemanager");
// Resource Reseach
exports.resourcesearch = require("oci-resourcesearch");
//Streaming
exports.streaming = require("oci-streaming");
// WAAS
exports.waas = require("oci-waas");
// Work Requests
exports.workrequests = require("oci-workrequests");
exports.cims = require("oci-cims");
exports.datasafe = require("oci-datasafe");
exports.bds = require("oci-bds");
exports.mysql = require("oci-mysql");
exports.nosql = require("oci-nosql");
exports.vault = require("oci-vault");
exports.secrets = require("oci-secrets");
exports.dataintegration = require("oci-dataintegration");
exports.ocvp = require("oci-ocvp");
exports.usageapi = require("oci-usageapi");
exports.blockchain = require("oci-blockchain");
exports.loganalytics = require("oci-loganalytics");
exports.logging = require("oci-logging");
exports.loggingingestion = require("oci-loggingingestion");
exports.loggingsearch = require("oci-loggingsearch");
exports.managementagent = require("oci-managementagent");
exports.managementdashboard = require("oci-managementdashboard");
exports.sch = require("oci-sch");
exports.cloudguard = require("oci-cloudguard");
exports.opsi = require("oci-opsi");
exports.computeinstanceagent = require("oci-computeinstanceagent");
exports.optimizer = require("oci-optimizer");
exports.tenantmanagercontrolplane = require("oci-tenantmanagercontrolplane");
exports.rover = require("oci-rover");
exports.databasemanagement = require("oci-databasemanagement");
exports.artifacts = require("oci-artifacts");
exports.apmcontrolplane = require("oci-apmcontrolplane");
exports.apmsynthetics = require("oci-apmsynthetics");
exports.apmtraces = require("oci-apmtraces");
exports.goldengate = require("oci-goldengate");
exports.networkloadbalancer = require("oci-networkloadbalancer");
exports.vulnerabilityscanning = require("oci-vulnerabilityscanning");
exports.databasemigration = require("oci-databasemigration");
exports.ailanguage = require("oci-ailanguage");
exports.operatoraccesscontrol = require("oci-operatoraccesscontrol");
exports.servicecatalog = require("oci-servicecatalog");
exports.bastion = require("oci-bastion");
exports.genericartifactscontent = require("oci-genericartifactscontent");
exports.jms = require("oci-jms");
exports.devops = require("oci-devops");
exports.aianomalydetection = require("oci-aianomalydetection");
exports.apmconfig = require("oci-apmconfig");
exports.datalabelingservice = require("oci-datalabelingservice");
exports.datalabelingservicedataplane = require("oci-datalabelingservicedataplane");
exports.waf = require("oci-waf");
exports.certificates = require("oci-certificates");
exports.certificatesmanagement = require("oci-certificatesmanagement");
exports.databasetools = require("oci-databasetools");
exports.servicemanagerproxy = require("oci-servicemanagerproxy");
exports.appmgmtcontrol = require("oci-appmgmtcontrol");
exports.identitydataplane = require("oci-identitydataplane");
exports.ospgateway = require("oci-ospgateway");
exports.visualbuilder = require("oci-visualbuilder");
exports.osubbillingschedule = require("oci-osubbillingschedule");
exports.osuborganizationsubscription = require("oci-osuborganizationsubscription");
exports.osubsubscription = require("oci-osubsubscription");
exports.osubusage = require("oci-osubusage");
exports.dashboardservice = require("oci-dashboardservice");
exports.aivision = require("oci-aivision");
exports.threatintelligence = require("oci-threatintelligence");
exports.aispeech = require("oci-aispeech");
exports.stackmonitoring = require("oci-stackmonitoring");
exports.servicemesh = require("oci-servicemesh");
exports.adm = require("oci-adm");
exports.licensemanager = require("oci-licensemanager");
exports.governancerulescontrolplane = require("oci-governancerulescontrolplane");
exports.onesubscription = require("oci-onesubscription");
exports.networkfirewall = require("oci-networkfirewall");
exports.emwarehouse = require("oci-emwarehouse");
exports.vnmonitoring = require("oci-vnmonitoring");
exports.fusionapps = require("oci-fusionapps");
exports.lockbox = require("oci-lockbox");
exports.mediaservices = require("oci-mediaservices");
exports.opa = require("oci-opa");
exports.opensearch = require("oci-opensearch");
exports.cloudbridge = require("oci-cloudbridge");
exports.cloudmigrations = require("oci-cloudmigrations");
exports.disasterrecovery = require("oci-disasterrecovery");
exports.containerinstances = require("oci-containerinstances");
exports.aidocument = require("oci-aidocument");
exports.queue = require("oci-queue");
exports.recovery = require("oci-recovery");
exports.vbsinst = require("oci-vbsinst");
exports.identitydomains = require("oci-identitydomains");
exports.accessgovernancecp = require("oci-accessgovernancecp");
exports.ocicontrolcenter = require("oci-ocicontrolcenter");
exports.osmanagementhub = require("oci-osmanagementhub");
exports.fleetsoftwareupdate = require("oci-fleetsoftwareupdate");
exports.computecloudatcustomer = require("oci-computecloudatcustomer");
exports.marketplacepublisher = require("oci-marketplacepublisher");
exports.redis = require("oci-redis");
exports.jmsjavadownloads = require("oci-jmsjavadownloads");
exports.psql = require("oci-psql");
exports.generativeai = require("oci-generativeai");
exports.generativeaiinference = require("oci-generativeaiinference");
exports.capacitymanagement = require("oci-capacitymanagement");
exports.globallydistributeddatabase = require("oci-globallydistributeddatabase");
exports.desktops = require("oci-desktops");
exports.emaildataplane = require("oci-emaildataplane");
exports.clusterplacementgroups = require("oci-clusterplacementgroups");
exports.marketplaceprivateoffer = require("oci-marketplaceprivateoffer");
exports.resourcescheduler = require("oci-resourcescheduler");
exports.demandsignal = require("oci-demandsignal");
exports.fleetappsmanagement = require("oci-fleetappsmanagement");
exports.delegateaccesscontrol = require("oci-delegateaccesscontrol");
exports.generativeaiagent = require("oci-generativeaiagent");
exports.generativeaiagentruntime = require("oci-generativeaiagentruntime");
exports.securityattribute = require("oci-securityattribute");
exports.zpr = require("oci-zpr");
exports.dblm = require("oci-dblm");
exports.mngdmac = require("oci-mngdmac");
exports.lustrefilestorage = require("oci-lustrefilestorage");
exports.modeldeployment = require("oci-modeldeployment");
exports.distributeddatabase = require("oci-distributeddatabase");
exports.apiaccesscontrol = require("oci-apiaccesscontrol");
exports.wlms = require("oci-wlms");
exports.dbmulticloud = require("oci-dbmulticloud");
exports.apiplatform = require("oci-apiplatform");
exports.multicloud = require("oci-multicloud");
