// oci-operatoraccesscontrol/index.js
"use strict";
/**
 * OperatorAccessControl API
 * Operator Access Control enables you to control the time duration and the actions an Oracle operator can perform on your Exadata Cloud@Customer infrastructure.
Using logging service, you can view a near real-time audit report of all actions performed by an Oracle operator.

Use the table of contents and search tool to explore the OperatorAccessControl API.

 * OpenAPI spec version: 20200630
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperatorControlAssignmentWaiter = exports.OperatorControlAssignmentClient = exports.OperatorControlWaiter = exports.OperatorControlClient = exports.OperatorActionsClient = exports.AccessRequestsWaiter = exports.AccessRequestsClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const accessrequests_waiter = __importStar(require("./lib/accessrequests-waiter"));
const operatorcontrol_waiter = __importStar(require("./lib/operatorcontrol-waiter"));
const operatorcontrolassignment_waiter = __importStar(require("./lib/operatorcontrolassignment-waiter"));
exports.AccessRequestsClient = client.AccessRequestsClient;
exports.AccessRequestsWaiter = accessrequests_waiter.AccessRequestsWaiter;
exports.OperatorActionsClient = client.OperatorActionsClient;
exports.OperatorControlClient = client.OperatorControlClient;
exports.OperatorControlWaiter = operatorcontrol_waiter.OperatorControlWaiter;
exports.OperatorControlAssignmentClient = client.OperatorControlAssignmentClient;
exports.OperatorControlAssignmentWaiter = operatorcontrolassignment_waiter.OperatorControlAssignmentWaiter;
