// oci-apigateway/index.js
"use strict";
/**
 * API Gateway API
 * API for the API Gateway service. Use this API to manage gateways, deployments, and related items.
For more information, see
[Overview of API Gateway](https://docs.oracle.com/iaas/Content/APIGateway/Concepts/apigatewayoverview.htm).

 * OpenAPI spec version: 20190501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestsWaiter = exports.WorkRequestsClient = exports.UsagePlansWaiter = exports.UsagePlansClient = exports.SubscribersWaiter = exports.SubscribersClient = exports.GatewayWaiter = exports.GatewayClient = exports.DeploymentWaiter = exports.DeploymentClient = exports.ApiGatewayWaiter = exports.ApiGatewayClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const apigateway_waiter = __importStar(require("./lib/apigateway-waiter"));
const deployment_waiter = __importStar(require("./lib/deployment-waiter"));
const gateway_waiter = __importStar(require("./lib/gateway-waiter"));
const subscribers_waiter = __importStar(require("./lib/subscribers-waiter"));
const usageplans_waiter = __importStar(require("./lib/usageplans-waiter"));
const workrequests_waiter = __importStar(require("./lib/workrequests-waiter"));
exports.ApiGatewayClient = client.ApiGatewayClient;
exports.ApiGatewayWaiter = apigateway_waiter.ApiGatewayWaiter;
exports.DeploymentClient = client.DeploymentClient;
exports.DeploymentWaiter = deployment_waiter.DeploymentWaiter;
exports.GatewayClient = client.GatewayClient;
exports.GatewayWaiter = gateway_waiter.GatewayWaiter;
exports.SubscribersClient = client.SubscribersClient;
exports.SubscribersWaiter = subscribers_waiter.SubscribersWaiter;
exports.UsagePlansClient = client.UsagePlansClient;
exports.UsagePlansWaiter = usageplans_waiter.UsagePlansWaiter;
exports.WorkRequestsClient = client.WorkRequestsClient;
exports.WorkRequestsWaiter = workrequests_waiter.WorkRequestsWaiter;
