// oci-ocvp/index.js
"use strict";
/**
 * Oracle Cloud VMware Solution API
 * Use the Oracle Cloud VMware API to create SDDCs and manage ESXi hosts and software.
For more information, see [Oracle Cloud VMware Solution](https://docs.oracle.com/iaas/Content/VMware/Concepts/ocvsoverview.htm).

 * OpenAPI spec version: 20230701
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestWaiter = exports.WorkRequestClient = exports.SddcWaiter = exports.SddcClient = exports.EsxiHostWaiter = exports.EsxiHostClient = exports.ClusterWaiter = exports.ClusterClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const cluster_waiter = __importStar(require("./lib/cluster-waiter"));
const esxihost_waiter = __importStar(require("./lib/esxihost-waiter"));
const sddc_waiter = __importStar(require("./lib/sddc-waiter"));
const workrequest_waiter = __importStar(require("./lib/workrequest-waiter"));
exports.ClusterClient = client.ClusterClient;
exports.ClusterWaiter = cluster_waiter.ClusterWaiter;
exports.EsxiHostClient = client.EsxiHostClient;
exports.EsxiHostWaiter = esxihost_waiter.EsxiHostWaiter;
exports.SddcClient = client.SddcClient;
exports.SddcWaiter = sddc_waiter.SddcWaiter;
exports.WorkRequestClient = client.WorkRequestClient;
exports.WorkRequestWaiter = workrequest_waiter.WorkRequestWaiter;
