// oci-dts/index.js
"use strict";
/**
 * Data Transfer Service API
 * Data Transfer Service API Specification
 * OpenAPI spec version: 1.0.017
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferPackageWaiter = exports.TransferPackageClient = exports.TransferJobWaiter = exports.TransferJobClient = exports.TransferDeviceWaiter = exports.TransferDeviceClient = exports.TransferApplianceEntitlementWaiter = exports.TransferApplianceEntitlementClient = exports.TransferApplianceWaiter = exports.TransferApplianceClient = exports.ShippingVendorsClient = exports.ApplianceExportJobWaiter = exports.ApplianceExportJobClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const applianceexportjob_waiter = __importStar(require("./lib/applianceexportjob-waiter"));
const transferappliance_waiter = __importStar(require("./lib/transferappliance-waiter"));
const transferapplianceentitlement_waiter = __importStar(require("./lib/transferapplianceentitlement-waiter"));
const transferdevice_waiter = __importStar(require("./lib/transferdevice-waiter"));
const transferjob_waiter = __importStar(require("./lib/transferjob-waiter"));
const transferpackage_waiter = __importStar(require("./lib/transferpackage-waiter"));
exports.ApplianceExportJobClient = client.ApplianceExportJobClient;
exports.ApplianceExportJobWaiter = applianceexportjob_waiter.ApplianceExportJobWaiter;
exports.ShippingVendorsClient = client.ShippingVendorsClient;
exports.TransferApplianceClient = client.TransferApplianceClient;
exports.TransferApplianceWaiter = transferappliance_waiter.TransferApplianceWaiter;
exports.TransferApplianceEntitlementClient = client.TransferApplianceEntitlementClient;
exports.TransferApplianceEntitlementWaiter = transferapplianceentitlement_waiter.TransferApplianceEntitlementWaiter;
exports.TransferDeviceClient = client.TransferDeviceClient;
exports.TransferDeviceWaiter = transferdevice_waiter.TransferDeviceWaiter;
exports.TransferJobClient = client.TransferJobClient;
exports.TransferJobWaiter = transferjob_waiter.TransferJobWaiter;
exports.TransferPackageClient = client.TransferPackageClient;
exports.TransferPackageWaiter = transferpackage_waiter.TransferPackageWaiter;
