// oci-dbmulticloud/index.js
"use strict";
/**
 * Oracle Database MultiCloud Data plane Integration
 * 1. Oracle Azure Connector Resource: This is for installing Azure Arc Server in ExaCS VM Cluster.
  There are two way to install Azure Arc Server (Azure Identity) in ExaCS VMCluster.
    a. Using Bearer Access Token or
    b. By providing Authentication token

2. Oracle Azure Blob Container Resource: This is for to capture Azure Container details
   and same will be used in multiple ExaCS VMCluster to mount the Azure Container.

3. Oracle Azure Blob Mount Resource: This is for to mount Azure Container in ExaCS VMCluster
   using Oracle Azure Connector and Oracle Azure Blob Container Resource.

 * OpenAPI spec version: 20240501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestWaiter = exports.WorkRequestClient = exports.OracleDbAzureVaultAssociationWaiter = exports.OracleDbAzureVaultAssociationClient = exports.OracleDbAzureVaultWaiter = exports.OracleDbAzureVaultClient = exports.OracleDbAzureKeyWaiter = exports.OracleDbAzureKeyClient = exports.OracleDBAzureConnectorWaiter = exports.OracleDBAzureConnectorClient = exports.OracleDBAzureBlobMountWaiter = exports.OracleDBAzureBlobMountClient = exports.OracleDBAzureBlobContainerWaiter = exports.OracleDBAzureBlobContainerClient = exports.MultiCloudResourceDiscoveryWaiter = exports.MultiCloudResourceDiscoveryClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const multicloudresourcediscovery_waiter = __importStar(require("./lib/multicloudresourcediscovery-waiter"));
const oracledbazureblobcontainer_waiter = __importStar(require("./lib/oracledbazureblobcontainer-waiter"));
const oracledbazureblobmount_waiter = __importStar(require("./lib/oracledbazureblobmount-waiter"));
const oracledbazureconnector_waiter = __importStar(require("./lib/oracledbazureconnector-waiter"));
const oracledbazurekey_waiter = __importStar(require("./lib/oracledbazurekey-waiter"));
const oracledbazurevault_waiter = __importStar(require("./lib/oracledbazurevault-waiter"));
const oracledbazurevaultassociation_waiter = __importStar(require("./lib/oracledbazurevaultassociation-waiter"));
const workrequest_waiter = __importStar(require("./lib/workrequest-waiter"));
exports.MultiCloudResourceDiscoveryClient = client.MultiCloudResourceDiscoveryClient;
exports.MultiCloudResourceDiscoveryWaiter = multicloudresourcediscovery_waiter.MultiCloudResourceDiscoveryWaiter;
exports.OracleDBAzureBlobContainerClient = client.OracleDBAzureBlobContainerClient;
exports.OracleDBAzureBlobContainerWaiter = oracledbazureblobcontainer_waiter.OracleDBAzureBlobContainerWaiter;
exports.OracleDBAzureBlobMountClient = client.OracleDBAzureBlobMountClient;
exports.OracleDBAzureBlobMountWaiter = oracledbazureblobmount_waiter.OracleDBAzureBlobMountWaiter;
exports.OracleDBAzureConnectorClient = client.OracleDBAzureConnectorClient;
exports.OracleDBAzureConnectorWaiter = oracledbazureconnector_waiter.OracleDBAzureConnectorWaiter;
exports.OracleDbAzureKeyClient = client.OracleDbAzureKeyClient;
exports.OracleDbAzureKeyWaiter = oracledbazurekey_waiter.OracleDbAzureKeyWaiter;
exports.OracleDbAzureVaultClient = client.OracleDbAzureVaultClient;
exports.OracleDbAzureVaultWaiter = oracledbazurevault_waiter.OracleDbAzureVaultWaiter;
exports.OracleDbAzureVaultAssociationClient = client.OracleDbAzureVaultAssociationClient;
exports.OracleDbAzureVaultAssociationWaiter = oracledbazurevaultassociation_waiter.OracleDbAzureVaultAssociationWaiter;
exports.WorkRequestClient = client.WorkRequestClient;
exports.WorkRequestWaiter = workrequest_waiter.WorkRequestWaiter;
