// oci-distributeddatabase/index.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistributedDbWorkRequestServiceWaiter = exports.DistributedDbWorkRequestServiceClient = exports.DistributedDbServiceWaiter = exports.DistributedDbServiceClient = exports.DistributedDbPrivateEndpointServiceWaiter = exports.DistributedDbPrivateEndpointServiceClient = exports.DistributedAutonomousDbServiceWaiter = exports.DistributedAutonomousDbServiceClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const distributedautonomousdbservice_waiter = __importStar(require("./lib/distributedautonomousdbservice-waiter"));
const distributeddbprivateendpointservice_waiter = __importStar(require("./lib/distributeddbprivateendpointservice-waiter"));
const distributeddbservice_waiter = __importStar(require("./lib/distributeddbservice-waiter"));
const distributeddbworkrequestservice_waiter = __importStar(require("./lib/distributeddbworkrequestservice-waiter"));
exports.DistributedAutonomousDbServiceClient = client.DistributedAutonomousDbServiceClient;
exports.DistributedAutonomousDbServiceWaiter = distributedautonomousdbservice_waiter.DistributedAutonomousDbServiceWaiter;
exports.DistributedDbPrivateEndpointServiceClient = client.DistributedDbPrivateEndpointServiceClient;
exports.DistributedDbPrivateEndpointServiceWaiter = distributeddbprivateendpointservice_waiter.DistributedDbPrivateEndpointServiceWaiter;
exports.DistributedDbServiceClient = client.DistributedDbServiceClient;
exports.DistributedDbServiceWaiter = distributeddbservice_waiter.DistributedDbServiceWaiter;
exports.DistributedDbWorkRequestServiceClient = client.DistributedDbWorkRequestServiceClient;
exports.DistributedDbWorkRequestServiceWaiter = distributeddbworkrequestservice_waiter.DistributedDbWorkRequestServiceWaiter;
