// oci-fleetappsmanagement/index.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FleetAppsManagementWorkRequestWaiter = exports.FleetAppsManagementWorkRequestClient = exports.FleetAppsManagementRunbooksWaiter = exports.FleetAppsManagementRunbooksClient = exports.FleetAppsManagementProvisionWaiter = exports.FleetAppsManagementProvisionClient = exports.FleetAppsManagementOperationsWaiter = exports.FleetAppsManagementOperationsClient = exports.FleetAppsManagementMaintenanceWindowWaiter = exports.FleetAppsManagementMaintenanceWindowClient = exports.FleetAppsManagementCatalogWaiter = exports.FleetAppsManagementCatalogClient = exports.FleetAppsManagementAdminWaiter = exports.FleetAppsManagementAdminClient = exports.FleetAppsManagementWaiter = exports.FleetAppsManagementClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const fleetappsmanagement_waiter = __importStar(require("./lib/fleetappsmanagement-waiter"));
const fleetappsmanagementadmin_waiter = __importStar(require("./lib/fleetappsmanagementadmin-waiter"));
const fleetappsmanagementcatalog_waiter = __importStar(require("./lib/fleetappsmanagementcatalog-waiter"));
const fleetappsmanagementmaintenancewindow_waiter = __importStar(require("./lib/fleetappsmanagementmaintenancewindow-waiter"));
const fleetappsmanagementoperations_waiter = __importStar(require("./lib/fleetappsmanagementoperations-waiter"));
const fleetappsmanagementprovision_waiter = __importStar(require("./lib/fleetappsmanagementprovision-waiter"));
const fleetappsmanagementrunbooks_waiter = __importStar(require("./lib/fleetappsmanagementrunbooks-waiter"));
const fleetappsmanagementworkrequest_waiter = __importStar(require("./lib/fleetappsmanagementworkrequest-waiter"));
exports.FleetAppsManagementClient = client.FleetAppsManagementClient;
exports.FleetAppsManagementWaiter = fleetappsmanagement_waiter.FleetAppsManagementWaiter;
exports.FleetAppsManagementAdminClient = client.FleetAppsManagementAdminClient;
exports.FleetAppsManagementAdminWaiter = fleetappsmanagementadmin_waiter.FleetAppsManagementAdminWaiter;
exports.FleetAppsManagementCatalogClient = client.FleetAppsManagementCatalogClient;
exports.FleetAppsManagementCatalogWaiter = fleetappsmanagementcatalog_waiter.FleetAppsManagementCatalogWaiter;
exports.FleetAppsManagementMaintenanceWindowClient = client.FleetAppsManagementMaintenanceWindowClient;
exports.FleetAppsManagementMaintenanceWindowWaiter = fleetappsmanagementmaintenancewindow_waiter.FleetAppsManagementMaintenanceWindowWaiter;
exports.FleetAppsManagementOperationsClient = client.FleetAppsManagementOperationsClient;
exports.FleetAppsManagementOperationsWaiter = fleetappsmanagementoperations_waiter.FleetAppsManagementOperationsWaiter;
exports.FleetAppsManagementProvisionClient = client.FleetAppsManagementProvisionClient;
exports.FleetAppsManagementProvisionWaiter = fleetappsmanagementprovision_waiter.FleetAppsManagementProvisionWaiter;
exports.FleetAppsManagementRunbooksClient = client.FleetAppsManagementRunbooksClient;
exports.FleetAppsManagementRunbooksWaiter = fleetappsmanagementrunbooks_waiter.FleetAppsManagementRunbooksWaiter;
exports.FleetAppsManagementWorkRequestClient = client.FleetAppsManagementWorkRequestClient;
exports.FleetAppsManagementWorkRequestWaiter = fleetappsmanagementworkrequest_waiter.FleetAppsManagementWorkRequestWaiter;
