// oci-objectstorage/index.js
"use strict";
/**
 * Object Storage Service API
 * Use Object Storage and Archive Storage APIs to manage buckets, objects, and related resources.
For more information, see [Overview of Object Storage](https://docs.oracle.com/iaas/Content/Object/Concepts/objectstorageoverview.htm) and
[Overview of Archive Storage](https://docs.oracle.com/iaas/Content/Archive/Concepts/archivestorageoverview.htm).

 * OpenAPI spec version: 20160918
 * Contact: opc_casper_users_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NodeFSBlob = exports.UploadManager = exports.ObjectStorageWaiter = exports.ObjectStorageClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const objectstorage_waiter = __importStar(require("./lib/objectstorage-waiter"));
exports.ObjectStorageClient = client.ObjectStorageClient;
exports.ObjectStorageWaiter = objectstorage_waiter.ObjectStorageWaiter;
const upload_manager_1 = require("./lib/upload-manager/upload-manager");
Object.defineProperty(exports, "UploadManager", { enumerable: true, get: function () { return upload_manager_1.UploadManager; } });
const node_fs_blob_1 = require("./lib/upload-manager/node-fs-blob");
Object.defineProperty(exports, "NodeFSBlob", { enumerable: true, get: function () { return node_fs_blob_1.NodeFSBlob; } });
