// oci-mysql/index.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestsWaiter = exports.WorkRequestsClient = exports.ReplicasWaiter = exports.ReplicasClient = exports.MysqlaasWaiter = exports.MysqlaasClient = exports.DbSystemWaiter = exports.DbSystemClient = exports.DbBackupsWaiter = exports.DbBackupsClient = exports.ChannelsWaiter = exports.ChannelsClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const channels_waiter = __importStar(require("./lib/channels-waiter"));
const dbbackups_waiter = __importStar(require("./lib/dbbackups-waiter"));
const dbsystem_waiter = __importStar(require("./lib/dbsystem-waiter"));
const mysqlaas_waiter = __importStar(require("./lib/mysqlaas-waiter"));
const replicas_waiter = __importStar(require("./lib/replicas-waiter"));
const workrequests_waiter = __importStar(require("./lib/workrequests-waiter"));
exports.ChannelsClient = client.ChannelsClient;
exports.ChannelsWaiter = channels_waiter.ChannelsWaiter;
exports.DbBackupsClient = client.DbBackupsClient;
exports.DbBackupsWaiter = dbbackups_waiter.DbBackupsWaiter;
exports.DbSystemClient = client.DbSystemClient;
exports.DbSystemWaiter = dbsystem_waiter.DbSystemWaiter;
exports.MysqlaasClient = client.MysqlaasClient;
exports.MysqlaasWaiter = mysqlaas_waiter.MysqlaasWaiter;
exports.ReplicasClient = client.ReplicasClient;
exports.ReplicasWaiter = replicas_waiter.ReplicasWaiter;
exports.WorkRequestsClient = client.WorkRequestsClient;
exports.WorkRequestsWaiter = workrequests_waiter.WorkRequestsWaiter;
